Server/src/Server.Domain/Entities/Book.cs

```diff
@@ -56,6 +56,14 @@ public void ComputeRating(ICollection<Review> reviews, double reviewRating)
         
         AverageRating = Math.Round((averageRating * reviewCount + reviewRating) / (reviewCount + 1), 2);
     }
+
+    public void ComputeCriticRating(ICollection<Review> reviews, double reviewRating)
+    {
+        var reviewCount = reviews.Count;
+        var averageCriticRating = AverageCriticRating ?? 0;
+
+        AverageCriticRating = Math.Round((averageCriticRating * reviewCount + reviewRating) / (reviewCount + 1), 2);
+    }
```

Comment:
nie rozumiem czemu to jest takie skomplikowane. Jeżeli mamy już wszystkie opinie to czemu nie możemy sobie wziąć średniej z `reviews` dodać `reviewRating` i podzielić? Wgl to ja bym to zrobił na poziomie bazy danych bo przy 100k opinii trochę by zmuliło

Reply:
Wzięcie sumy to raczej tylko spowolni, musisz pobrać sume i count wszystkich. 
Teraz bierzemy tylko count, mnożymy to przez średnia, dodajemy nowa, dzielimy itd. 
Co do tego żeby to zrobić po stronie bazy to zgadzam się, byłoby szybsze, tylko pytanie ile zajmie ogarniecie tego dla każdego przypadku (dodanie, edycja, usunięcie)

Reply:
Przecież po to masz średnia, żeby ją podzielić przez liczbę opinii, robienie tu sumy wszystkich po kolei jest raczej totalnie bez sensu.

Reply:
to dlaczego nie bierzemy samego `count` tych opinii? Mi chodzi o sam fakt że pobieramy literalnie wszystkie opinie (i jeszcze użytkowników do nich) po to żeby później wgl z tego nie skorzystać. Chodzi mi o to https://github.com/DontEatRice/BookBook/blob/a4f1ac9956544c052936c1bcfcd8c9f2a183b679/Server/src/Server.Application/CommandHandlers/User/AddReviewHandler.cs#L43

Reply:
Tak, to wiem. Zmienię za chwilę.

Reply:
Poprawiłem, możesz zerknąć. Tak naprawdę to pewnie w wielu miejscach można by poprawić performance (chociażby trzymając Id klucza obcego, a nie tylko obiekt. Popatrzę na inne, może się znajdzie coś na szybko do poprawienia.